op-supervisor/supervisor/backend/db/fromda/db.go

```diff
@@ -137,6 +137,19 @@ func (db *DB) Invalidated() (pair types.DerivedBlockSealPair, err error) {
 	}, nil
 }
 
+func (db *DB) Replacement(derived eth.BlockID) error {
+	db.rwLock.RLock()
+	defer db.rwLock.RUnlock()
+	_, link, err := db.derivedNumToFirstSource(derived.Number)
```

Comment:
We shouldn't add APIs that take a `eth.BlockID` argument and only use the block-number, since that falsely implies that the block-hash is being checked, which it is not here. Let's check that the link has a block-hash that matches the derived block hash.

Reply:
I think this is a misleading parameter name that I should change -- If a caller is asking if a block is a replacement block, they may not yet have the correct derived, and so we shouldn't expect it to match. The caller has likely derived an invalid block and is now attempting to find out if the item which *is* in the database (which doesn't match) is a replacement.

Since we can't actually do a block check here, I'll reframe the API to `blockNum uint64`

Reply:
We can do a block check here, the L2 blockhash is part of the derived side of the link. Doing the blockhash check can remove surprises from the caller, by making the check not report an invalid thing when there's some reorg.